### PR TITLE
Fixes for the random forest processes

### LINKED ISSUE #349: `predictors` vector cube in fit_class_random_forest
https://github.com/Open-EO/openeo-processes/blob/e4df864846e833dda4669e190669a2db5b365c62/proposals/fit_class_random_forest.json#L12

If I understand the ML aspects of fit_class_random_forest well enough I think the description should mention that the `predictors` vector cube should be a 2D data cube: 
- one dimension being the "geometry" dimension
- the other dimension being the "ML feature" dimension

### LINKED ISSUE #350: default for train/validation split in fit_class_random_forest
https://github.com/Open-EO/openeo-processes/blob/e4df864846e833dda4669e190669a2db5b365c62/proposals/fit_class_random_forest.json#L27-L33

Maybe this has been discussed before, but can't we pick a sensible default for the training-validation split? e.g. 80% / 20%?

### PR DESCRIPTION
**Pull request by m-mohr**
Fixes #350, #339, #354, #355 - for context please see the corresponding issues.

### CODE REVIEW

File: proposals/fit_class_random_forest.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "fit_class_random_forest",
     "summary": "Train a random forest classification model",
-    "description": "Executes the fit of a random forest classification based on the user input of target and predictors. The Random Forest classification model is based on the approach by Breiman (2001).",
+    "description": "Executes the fit of a random forest classification based on the user input of target and predictors. The Random Forest classification model is based on the approach by Breiman (2001). Training/validation split must be done before.",
```
[the remark this answers is not in the document]
Comment (mattia6690) [reply proposals/fit_class_random_forest.json:4]: This applies also to the fit_regr_random_forest process

[the remark this answers is not in the document]
Comment (m-mohr) [reply proposals/fit_class_random_forest.json]: Done

File: proposals/fit_regr_random_forest.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "fit_regr_random_forest",
     "summary": "Train a random forest regression model",
-    "description": "Executes the fit of a random forest regression based on the user input of target and predictors. The Random Forest regression model is based on the approach by Breiman (2001).",
+    "description": "Executes the fit of a random forest regression based on the user input of target and predictors. The Random Forest regression model is based on the approach by Breiman (2001). Training/validation split must be done before.",
```
Comment (soxofaan) [inline proposals/fit_regr_random_forest.json]: see my comment above about mentioning "validation split" at all.

File: proposals/fit_class_random_forest.json [hunk 3]
```diff
@@ -24,13 +24,23 @@
...
         },
         {
-            "name": "training",
-            "description": "The amount of training data to be used in the classification, given as a fraction. The sampling will be chosen randomly through the data object. The remaining data will be used as test data for the validation.",
-            "schema": {
-                "type": "number",
-                "exclusiveMinimum": 0,
-                "maximum": 1
-            }
+            "name": "max_variables",
```
[the remark this answers is not in the document]
Comment (m-mohr) [reply proposals/fit_class_random_forest.json:27]: Then you'd get different behavior across back-ends and end up with an undefined default (auto?) that you later need to still handle. I don't think this is a good idea to leave it unspecified. We can remove the parameter, but then we have to specify the default behavior for the process. I'm not sure we can agree on this so I'd be inclined to leave it as it is for now and just let a user choose from the options the back-ends provide. Back-ends surely need to adapt the process description to their capabilities.

File: proposals/fit_class_random_forest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-mohr) [reply proposals/fit_class_random_forest.json]: @soxofaan It is already "weakened" as proposed by @mattia6690 . Is that good enough?

File: proposals/fit_class_random_forest.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (soxofaan) [reply proposals/fit_class_random_forest.json:27]: > you'd get different behavior across back-ends 

You will get different results anyway. Each implementation has it's own tuning/hyper parameters and  there is so much "random walking" in ML training you can not avoid that I'm afraid

File: proposals/fit_class_random_forest.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (soxofaan) [reply proposals/fit_class_random_forest.json]: It's indeed weaker now, but to me "The process does not include a separate split of the data in test and training data."  sounds redundant if you already state ".. classification based on training data." in the previous sentence.

But that's just personal I guess. Don't let that be a blocker for proceeding here.

File: proposals/fit_class_random_forest.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (soxofaan) [reply proposals/fit_class_random_forest.json:27]: I'm just suggesting to drop the parameter for now.
I think we can still define/recommend a behavior if that is important (e.g. `sqrt` seems to be available in all listed implementations at https://github.com/Open-EO/openeo-processes/issues/355#issuecomment-1076450734)

[the remark this answers is not in the document]
Comment (m-mohr) [reply proposals/fit_class_random_forest.json:27]: Created issue #358 so that I can merge for the deliverables.

**Pull request merged**